### PR TITLE
Fix ignore_duplicates missing filtered tracks

### PR DESCRIPTION
**Pull request by gavtroy**
Really just checking for duplicates again when unfiltering. A bit messy but it does the job.

The "entryp" naming is consistent with `hash_remove()` etc.

Relates to #423.

### CODE REVIEW

**Comment by flyingmutant** [comment]
This makes code and intent really hard to follow: we are modifying hash (in a non-obvious way) inside a function called `hash_add_to_views`. I'd say some further work on this is required.

**Comment by gavtroy** [comment]
Then for simplicity we can leave the duplicates in the hash and just skip adding them to the views. The only slight downside is future `hash_add_to_views` invocations (i.e. clearing a filter) could be slightly slower with a lot of duplicates... not something that matters in practice.
